cmd/controller-manager/app/controllermanager.go

```diff
@@ -124,22 +140,33 @@ func startControllers(
 			continue
 		}
 
-		err := initFn(ctx, controllerCtx)
+		controller, err := initFn(ctx, controllerCtx)
 		if err != nil {
 			return fmt.Errorf("error starting %q: %w", controllerName, err)
 		}
 		klog.Infof("Started %q", controllerName)
+
+		healthCheckHandler.AddReadyzChecker(controllerName, controllermanager.HealthzCheckerAdaptor(controllerName, controller))
```

Comment:
Do we need to capture the controller name in the error message? Would a function literal that calls `controller.IsControllerReady` work?

Reply:
using a function literal is fine as well